docs/testplan/ecn/WRED-and-ECN-statistics-testplan.md

```diff
@@ -0,0 +1,254 @@
+ | counterpoll wred-port enable/disable                             |  Enable/Disable the port level counters           |
+ | counterpoll wred-queue interval &lt;value&gt;                    |  Set polling interval for queue level counters           |
+ | counterpoll wred-port interval &lt;value&gt;                     |  Set polling interval for port level counters          |
+ | **Show commands**                                                |
+ | show queue wredcounters [interface-name]                         |  Display the statistics on the console            |
+ | show interfaces counters detailed Ethernet&lt;&gt;               |  Display the statistics on the console           |
+ | sonic-clear queue wredcounters                                   |  Statistics are cleared on user request            |
+ | sonic-clear counters                                             |  Statistics are cleared on user request           |
+ | counterpoll show                                                 |  Display the status of the counters           |
+ ### **Setup configuration**
+ -------------------
+ PTF setup: T0, T1
+ ### **Related DUT configuration files**
+ -----------------------------------
+       "WRED_PROFILE": {
+          "AZURE_LOSSY": {
+             "ecn": "ecn_all",
+             "green_drop_probability": "100",
+             "green_max_threshold": "2240000",
+             "green_min_threshold": "1120000",
+             "red_drop_probability": "100",
+             "red_max_threshold": "1344000",
+             "red_min_threshold": "672000",
+             "wred_green_enable": "true",
+             "wred_red_enable": "true",
+             "wred_yellow_enable": "true",
+             "yellow_drop_probability": "100",
+             "yellow_max_threshold": "1792000",
+             "yellow_min_threshold": "896000"
+         }
+      }
+     "QUEUE": {
+         "Ethernet252|0": {
+             "wred_profile": "[WRED_PROFILE|AZURE_LOSSY]"
+         }
+     }
+ ### **Related SAI APIs and Attributes**
+ -------------------------------
+ Following SAI statistics will get tested:
+
+ SAI_PORT_STAT_GREEN_WRED_DROPPED_PACKETS <br>
+ SAI_PORT_STAT_YELLOW_WRED_DROPPED_PACKETS <br>
+ SAI_PORT_STAT_RED_WRED_DROPPED_PACKETS <br>
+ SAI_PORT_STAT_WRED_DROPPED_PACKETS <br>
+ SAI_QUEUE_STAT_WRED_ECN_MARKED_PACKETS <br>
+ SAI_QUEUE_STAT_WRED_ECN_MARKED_BYTES <br>
+
+ <span id="_Toc463421033" class="anchor"></span>
+ ## **Test methodology**
+
+ 1. Apply WRED/ECN Profile to the queue  .
+
+ 2. Query STATE_DB using REDIS CLI to find out if platform supports WRED/ECN  .
+
+ 3. On supported platform:
+
+    Enable/Disable WRED at queue level using config:<br>
+        counterpoll wred-queue enable/disable<br>
+    Enable/Disable WRED at interface level using config:
+        counterpoll wred-port enable/disable<br>
+
+ 4. on egress port  set   attribute SAI_PORT_ATTR_PKT_TX_ENABLE = false . <br>
```

Comment:
Just to confirm, you are going to use RPC image to do the test, right?
In that case, the way to close the egress port differs among vendors. It should be fine if you leverage the existing interface [sai_thrift_port_tx_enable](https://github.com/sonic-net/sonic-mgmt/blob/master/tests/saitests/py3/sai_base_test.py#L181) and [sai_thrift_port_tx_disable](https://github.com/sonic-net/sonic-mgmt/blob/master/tests/saitests/py3/sai_base_test.py#L209) which already tolerate the difference.